src/app/core/components/left-nav/left-nav-group-datasource.ts

```diff
@@ -1,36 +1,61 @@
 import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
 import { GroupInfo } from 'src/app/shared/models/content/group-info';
-import { GroupNavigationService, NavMenuGroup } from '../../http-services/group-navigation.service';
+import { GroupNavigationService, GroupNavigationChild, GroupNavigationData } from '../../http-services/group-navigation.service';
+import { NavTreeElement } from '../../models/left-nav-loading/nav-tree-data';
 import { LeftNavDataSource } from './left-nav-datasource';
 
-export class LeftNavGroupDataSource extends LeftNavDataSource<GroupInfo, NavMenuGroup> {
+export class LeftNavGroupDataSource extends LeftNavDataSource<GroupInfo, NavTreeElement> {
 
   constructor(
     private groupNavigationService: GroupNavigationService,
   ) {
     super();
   }
 
-  addDetailsToTreeElement(contentInfo: GroupInfo, treeElement: NavMenuGroup): NavMenuGroup {
+  addDetailsToTreeElement(contentInfo: GroupInfo, treeElement: NavTreeElement): NavTreeElement {
     return contentInfo.title ? { ...treeElement, title: contentInfo.title } : { ...treeElement };
   }
 
-  fetchRootTreeData(): Observable<NavMenuGroup[]> {
-    return this.groupNavigationService.getRoot();
+  fetchRootTreeData(): Observable<NavTreeElement[]> {
+    return this.groupNavigationService.getRoot().pipe(
+      map(groups => groups.map(mapChild))
+    );
   }
 
-  fetchNavDataFromChild(id: string, _child: GroupInfo): Observable<{ parent: NavMenuGroup, elements: NavMenuGroup[] }> {
-    return this.groupNavigationService.getNavData(id).pipe(
-      map(data => ({ parent: data.parent, elements: data.groups }))
+  fetchNavDataFromChild(id: string, _child: GroupInfo): Observable<{ parent: NavTreeElement, elements: NavTreeElement[] }> {
+    return this.groupNavigationService.getGroupNavigation(id).pipe(
+      map(mapNavData)
     );
   }
 
-  fetchNavData(el: NavMenuGroup): Observable<{ parent: NavMenuGroup, elements: NavMenuGroup[] }> {
-    return this.groupNavigationService.getNavData(el.id).pipe(
-      map(data => ({ parent: data.parent, elements: data.groups }))
+  fetchNavData(el: NavTreeElement): Observable<{ parent: NavTreeElement, elements: NavTreeElement[] }> {
+    return this.groupNavigationService.getGroupNavigation(el.id).pipe(
+      map(mapNavData)
     );
   }
 
+}
+
+function mapChild(child: GroupNavigationChild): NavTreeElement {
+  return {
+    id: child.id,
+    title: child.name,
+    hasChildren: child.type !== 'User',
```

Comment:
I don't get that condition, a group might have no children, even if it's not a user right ?
I feel like I'm missing something, what does `hasChildren` reflects for a `NavTreeElement` ?

Edit:
I think I get it: it means that the node _might_ have children, right ?

Reply:
Out of scope of the PR as well (just a move of the previous same logic to here).
Actually, the current group navigation service does not return whether the group has children or not, so I just return if it might have children. But anyway, the template does not do anything with this info for now for group I think.